libraries/conf_template.rb

```diff
@@ -0,0 +1,85 @@
+# coding: UTF-8
+#
+# Cookbook Name:: cerner_splunk
+# File Name:: conf_template.rb
+
+module CernerSplunk
+  # Module for generating template data
+  module ConfTemplate
+    # Evaluate a Proc (or a Proc returning Procs)
+    def self.collapse_proc(initial, depth: 10, arguments: {}, label: nil)
```

Comment:
Does the depth ever exceed more than 3-4 at this point?  curious if there was a specific reason for putting it at 10.